Modules/signalmodule.c

```diff
@@ -244,6 +244,16 @@ trip_signal(int sig_num)
 
     Handlers[sig_num].tripped = 1;
 
+    if (!is_tripped) {
+        /* Set is_tripped after setting .tripped, as it gets
+           cleared in PyErr_CheckSignals() before .tripped. */
+        is_tripped = 1;
+        Py_AddPendingCall(checksignals_witharg, NULL);
+    }
+
+    /* And then write to the wakeup fd *after* setting all the globals and
+       doing the Py_AddPendingCall (bpo-30038) */
```

Comment:
I would like to see here the rationale for setting is_tripped before writing into the wake up fd. You already wrote it in the http://bugs.python.org/issue30038 and in the PR description. Just copy it here please ;-)